get-started/index.md

```diff
@@ -62,7 +65,291 @@ the specifics of how to deploy your Cryostat instance. Continue to [Setup](#setu
+
+```yaml
+apiVersion: v1
+kind: Service
+...
+spec:
+  ports:
+    - name: "cryostat-agent"
+      port: 9977
+      targetPort: 9977
+...
+```
+
+More details about the configuration options for the Cryostat Agent [are available here](https://github.com/cryostatio/cryostat-agent/blob/main/README.md#configuration).
+
+#### [Using JMX](#using-jmx)
+Cryostat is also able to use Java Management Extensions (JMX) to communicate with target applications. This is a standard JDK feature that can be enabled by passing JVM
+flags to your application at startup. A basic and insecure setup suitable for testing requires only the following three flags:
+
+```
+-Dcom.sun.management.jmxremote.port=9091
+-Dcom.sun.management.jmxremote.ssl=false
+-Dcom.sun.management.jmxremote.authenticate=false
+```
+
+[comment]: # TODO explain how to configure SSL and auth for JMX, or link to external docs
+
+In a real scenario you should enable both SSL and authentication on your application. You can then [trust the certificate](/guides/#add-a-trusted-certificate)
+and [store the credentials](/guides/#store-jmx-credentials).
+
+Depending on your application or its framework, you may set these flags directly in a `Dockerfile` entrypoint, an environment variable, or similar. This may or
+may not require a container image rebuild, and it will require the container to be restarted. Once this is done the application container will be listening for
+incoming JMX connections on port 9091. Let's assume it can be done by setting an environment variable, so we only need to mofify our Deployment:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+...
+spec:
+  ...
+  template:
+    ...
+    spec:
+      containers:
+        - name: sample-app
+          image: docker.io/myorg/myapp:latest
+          env:
+            - name: JAVA_OPTS
+              value: "-Dcom.sun.management.jmxremote.port=9091 -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.authenticate=false"
+            ...
+```
+
+Next, we need to configure an OpenShift Service to expose this port for cluster-internal traffic, so that Cryostat can see
+and connect to this application JMX port.
+
+```yaml
+apiVersion: v1
+kind: Service
+...
+spec:
+  ports:
+    - name: "jfr-jmx"
+      port: 9091
+      targetPort: 9091
+...
+```
+
+Cryostat queries the OpenShift API server and looks for Services with a port either named `jfr-jmx` or with the number `9091`. One or both of these conditions
+must be met or else Cryostat will not automatically detect your application. In this case you may wish to use the [Cryostat Agent](#using-the-cryostat-agent-with-jmx)
+to enable discovery, while keeping communications over JMX rather than HTTP.
+
+#### [Using the Cryostat Agent with JMX](#using-the-cryostat-agent-with-jmx)
+The two prior sections have discussed how to use the Cryostat Agent to do application discovery and expose data over HTTP, and how to use OpenShift configurations
```

Comment:
How about formatting this paragraph to be easy to read?

```diff
 #### [Using the Cryostat Agent with JMX](#using-the-cryostat-agent-with-jmx)
-The two prior sections have discussed how to use the Cryostat Agent to do application discovery and expose data over HTTP, and how to use OpenShift configurations
-for discovery and JMX to expose data. There is a third, hybrid approach: using the Cryostat Agent to do application discovery, and JMX to expose data. This may be
+The two prior sections have discussed:
+  - How to use the Cryostat Agent to do application discovery and expose data over HTTP.
+  - How to use `Service` configurations for discovery and JMX to expose data. 
+
+There is a third, hybrid approach: **using the Cryostat Agent to do application discovery, and JMX to expose data**. This may be
 useful since the Agent HTTP data model is readonly, whereas JMX is read-write. This means that using JMX to communicate between Cryostat and your applications
-allows for more dynamic flexibility, for example the ability to start and stop Flight Recordings on demand. Using the Cryostat Agent for application discovery
-is also more flexible than depending on OpenShift Services with specially-named or specially-numbered ports. For more context about these concepts, please review
+allows for more dynamic flexibility, for example, the ability to start and stop Flight Recordings on demand. Using the Cryostat Agent for application discovery
+is also more flexible than depending on `Services` with specially-named or specially-numbered ports. 
+
+For more context about these concepts, please review
 the previous two sections on [using the Cryostat Agent](#using-the-cryostat-agent) and [using JMX](#using-jmx).
```

![Screenshot from 2023-04-26 17-04-05](https://user-images.githubusercontent.com/68053619/234702232-679496d4-ca9c-48e0-bd86-abd39e44bc0a.png)